var/spack/repos/builtin/packages/espressopp/package.py

```diff
@@ -0,0 +1,81 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+
+from spack import *
+
+
+class Espressopp(Package):
+    """ESPResSo++ is an extensible, flexible, fast and parallel simulation
+       software for soft matter research. It is a highly versatile software
+       package for the scientific simulation and analysis of coarse-grained
+       atomistic or bead-spring models as they are used in soft matter research
+    """
+    homepage = "https://espressopp.github.io"
+    url      = "https://github.com/espressopp/espressopp/archive/v1.9.4.zip"
+
+    version('develop', git='https://github.com/espressopp/espressopp.git', branch='master')
+    version('1.9.4', git='https://github.com/espressopp/espressopp.git', tag='v1.9.4')
+
+    variant('debug', default=False, description='Build debug version')
+    variant('ug', default=False, description='Build user guide')
+    variant('pdf', default=False, description='Build user guide in pdf format')
+    variant('dg', default=False, description='Build developer guide')
+    variant('tests', default=False, description='Run the tests')
+
+    depends_on("cmake@3.7.1:", type='build')
+    depends_on("mpi", type=('build', 'link', 'run'))
+    depends_on("boost+serialization+filesystem+system+python+mpi", when='@1.9.4:', type=('build', 'link', 'run'))
```

Comment:
@citibeth I of course agree, but I am not sure this is the intended usage, from docu

> “run”: the project is used by the project at runtime. The package will be added to PATH and PYTHONPATH.

so it looks like it will matter only for `spack load espressopp`, but no packages so far declare `mpi` as run dependency. 

@tgamblin @mathstuf : don't remember who added dependency types, but could you comment on `run` for `mpi`?